subsys/logging/Kconfig.backends

```diff
@@ -17,44 +17,16 @@ config LOG_BACKEND_UART_ASYNC
 	depends on UART_ASYNC_API
 	depends on !LOG_BACKEND_UART_OUTPUT_DICTIONARY_HEX
 
-config LOG_BACKEND_UART_OUTPUT_DICTIONARY
-	bool
-	depends on LOG2
-	select LOG_DICTIONARY_SUPPORT
-	help
-	  UART backend is in dictionary-based logging output mode.
-
-choice LOG_BACKEND_UART_OUTPUT
-	prompt "UART Backend Output Mode"
-	default LOG_BACKEND_UART_OUTPUT_TEXT
-
-config LOG_BACKEND_UART_OUTPUT_TEXT
-	bool "Text"
-	help
-	  Output in text.
+backend = UART
+backend-str = uart
+source "subsys/logging/Kconfig.template.log_format_config"
 
-config LOG_BACKEND_UART_SYST_ENABLE
-	bool "MIPI SyS-T"
-	depends on LOG_BACKEND_UART
-	depends on LOG_MIPI_SYST_ENABLE
-	help
-	  When enabled backend is using UART to output syst format logs.
-
-config LOG_BACKEND_UART_OUTPUT_DICTIONARY_HEX
+config LOG_BACKEND_$(backend)_OUTPUT_DICTIONARY_HEX
 	bool "Dictionary (hexadecimal)"
 	depends on LOG2
-	select LOG_BACKEND_UART_OUTPUT_DICTIONARY
-	help
-	  Dictionary-based logging output in hexadecimal.
-
-config LOG_BACKEND_UART_OUTPUT_DICTIONARY_BIN
-	bool "Dictionary (binary)"
-	depends on LOG2
-	select LOG_BACKEND_UART_OUTPUT_DICTIONARY
+	select LOG_BACKEND_$(backend)_OUTPUT_DICTIONARY
 	help
-	  Dictionary-based logging output in binary.
-
-endchoice
+	  Dictionary-based logging output in hexadecimal. Supported only for UART backend.
```

Comment:
This presents an option `LOG_BACKEND_UART_OUTPUT_DICTIONARY_HEX` in addition to the output choice block in the template. Note that `..._BIN` is enclosed in a `choice`/`endchoice` pair, but `..._HEX` is outside the choice block. So even if `..._BIN` is enabled (i.e. selected in the choice block), `..._HEX` can be set independently. But in reality, these two options are mutually exclusive.

Reply:
@dcpleung Corrected this. Thanks for noticing this.